packages/cli/src/cli/overview/tile.generator.ts

```diff
@@ -0,0 +1,122 @@
+const tilerGoogle = new Tiler(GoogleTms);
+
+function getTiler(tileMatrix: string): { tiler: Tiler; tileMatrix: TileMatrixSet } {
+  if (tileMatrix === GoogleTms.identifier) return { tiler: tilerGoogle, tileMatrix: GoogleTms };
+  else if (tileMatrix === Nztm2000QuadTms.identifier || tileMatrix === Nztm2000Tms.identifier)
+    return { tiler: tilerNZTM2000Quad, tileMatrix: Nztm2000QuadTms };
+  else throw new Error(`Invalid Tile Matrix provided ${tileMatrix}`);
+}
+
+export interface JobTiles {
+  path: string;
+  tileMatrix: string;
+  tiles: string[];
+  files: NamedBounds[];
+}
+
+const Q = pLimit(2);
+
+let count = 0;
+let skipped = 0;
+
+export type RpcContract = {
+  tile(jobTiles: JobTiles): Promise<void>;
+};
+
+const worker = new WorkerRpc<RpcContract>({
+  async tile(jobTiles: JobTiles): Promise<void> {
+    const logger = LogConfig.get().child({ workerId: worker.id, messageId: worker.messageCount });
+    logger.info({ count, skipped }, 'TaskCount');
+
+    let lastTime = performance.now();
+    const todo = jobTiles.tiles.map((qk) => {
+      return Q(async () => {
+        const tile = QuadKey.toTile(qk);
+        count++;
+        if (count % 100 === 0) {
+          const duration = performance.now() - lastTime;
+          lastTime = Number(performance.now().toFixed(4));
+          logger.info({ count, total: jobTiles.tiles.length, duration }, 'Progress');
+        }
+
+        const outputTile = `tiles/${tile.z}/${tile.x}/${tile.y}.webp`;
+        const outputFile = fsa.join(jobTiles.path, outputTile);
+        const exists = await fsa.exists(outputFile);
+        if (exists) {
+          skipped++;
+          return;
+        }
+        const buffer = await getComposedTile(jobTiles, tile);
+        if (buffer != null) await fsa.write(outputFile, buffer);
+      });
+    });
+
+    await Promise.all(todo);
+  },
+});
+
+if (parentPort) worker.bind(parentPort);
+
+async function getComposedTile(jobTiles: JobTiles, tile: Tile): Promise<Buffer | undefined> {
+  const files = jobTiles.files;
+  const tiffPaths: string[] = [];
+  const { tiler, tileMatrix } = getTiler(jobTiles.tileMatrix);
+  const tileBounds = tileMatrix.tileToSourceBounds(tile);
+  for (const c of files) {
+    if (!tileBounds.intersects(Bounds.fromJson(c))) continue;
+    const tiffPath = c.name;
+    tiffPaths.push(tiffPath);
+  }
+
+  const todoTiffs: Promise<CogTiff>[] = [];
+  for (const tiffPath of tiffPaths) {
+    const tiff = CoSources.getCog(tiffPath);
+    todoTiffs.push(tiff);
+  }
+
+  const tiffs = await Promise.all(todoTiffs);
+  const layers = await tiler.tile(tiffs, tile.x, tile.y, tile.z);
+  if (layers.length === 0) return;
+  const res = await TileComposer.compose({
+    layers,
+    format: ImageFormat.Webp,
+    background: DefaultBackground,
+    resizeKernel: DefaultResizeKernel,
+  });
+  if (res.layers === 0) return;
+
+  // Check and skip if the buffer is empty webp
+  if (res.buffer.byteLength < 215) {
```

Comment:
Thinking about this more, we could run a `libvips`  `stats` function to see if there is any non alpha in the image.

I am unsure if we are creating other empty tiles or if all empty tiles will always create this buffer.

Reply:
https://sharp.pixelplumbing.com/api-input#stats

Reply:
I have tried to skip the ones with `stat.isOpaque`; Seems it still pick the empty ones. I think this might be the empty images come from which already have a background in it. https://github.com/linz/basemaps/blob/c232694ad1f1186b7b56a8af1def56ef165df8fd/packages/tiler-sharp/src/index.ts#L100

Reply:
This the stat for empty images. We might need a smart why to check this? 
`{
  channels: [
    {
      min: 0,
      max: 0,
      sum: 0,
      squaresSum: 0,
      mean: 0,
      stdev: 0,
      minX: 0,
      minY: 0,
      maxX: 0,
      maxY: 0
    },
    {
      min: 0,
      max: 0,
      sum: 0,
      squaresSum: 0,
      mean: 0,
      stdev: 0,
      minX: 0,
      minY: 0,
      maxX: 0,
      maxY: 0
    },
    {
      min: 0,
      max: 0,
      sum: 0,
      squaresSum: 0,
      mean: 0,
      stdev: 0,
      minX: 0,
      minY: 0,
      maxX: 0,
      maxY: 0
    },
    {
      min: 0,
      max: 0,
      sum: 0,
      squaresSum: 0,
      mean: 0,
      stdev: 0,
      minX: 0,
      minY: 0,
      maxX: 0,
      maxY: 0
    }
  ],
  isOpaque: false,
  entropy: 0,
  sharpness: 0,
  dominant: { r: 8, g: 8, b: 8 }
}`

Reply:
Maybe we check if all the channels.toString equal to `{ min: 0, max: 0, sum: 0, squaresSum: 0, mean: 0, stdev: 0, minX: 0, minY: 0, maxX: 0, maxY: 0 }` and skip it.

Reply:
Hmm, the last channel should be the alpha channel, and it should be 255 (?) for full alpha 

so something like  `image[image.bands - 1].min == 255`  should work?

Reply:
For this empty ones, it looks like all four channels are the same, they all be `{ min: 0, max: 0, sum: 0, squaresSum: 0, mean: 0, stdev: 0, minX: 0, minY: 0, maxX: 0, maxY: 0 }`

Reply:
then maybe alpha of 0? is full alpha? `image[image.bands - 1].max == 0` ?

Reply:
I will try to print all `image[image.bands - 1].max == 0` ones and see if they are all empty channels.

Reply:
Looks like working. I have updated that.